server/__init__.py

```diff
@@ -17,14 +17,37 @@
 #  limitations under the License.
 ###############################################################################
 
+from girder import events
 from girder.constants import AccessType
 from girder.utility.model_importer import ModelImporter
 
 from .rest import TilesItemResource
 
 
+def _postUpload(event):
+    """
+    Called when a file is uploaded. We check the parent item to see if it is
+    expecting a large image upload, and if so we register this file as the
+    result image.
+    """
+    file = event.info['file']
+    if 'itemId' not in file:
+        return
+
+    itemModel = ModelImporter.model('item')
+    item = itemModel.load(file['itemId'], force=True, exc=True)
+
+    if item.get('expectedLargeImage') and (file['name'].endswith('.tiff') or
+            file.get('mimeType') == 'image/tiff'):
```

Comment:
If the file doesn't actually have its ``mimeType`` field set to ``image/tiff`` here, we need to set it and update the file.

Logic in the POST endpoint to short-circuit the conversion process depends on this field being correct for TIFF files.